executor/common_linux.h

```diff
@@ -5093,17 +5103,22 @@ static void setup_sysctl()
 		// (sshd or another random test process).
 		{"/proc/sys/vm/oom_kill_allocating_task", "1"},
 		// This blocks some of the ways the fuzzer can trigger a reboot.
-		// ctrl-alt-del=0 tells kernel to signal cad_pid instead of rebooting
-		// and setting cad_pid to the current pid (transient "syz-executor setup") makes it a no-op.
+		// ctrl-alt-del=0 tells kernel to signal cad_pid instead of rebooting.
+		// We set cad_pid to a transient process pid ctrl-alt-del a no-op.
+		// Note: we need to write a live process pid.
 		// For context see: https://groups.google.com/g/syzkaller-bugs/c/WqOY4TiRnFg/m/6P9u8lWZAQAJ
 		{"/proc/sys/kernel/ctrl-alt-del", "0"},
-		{"/proc/sys/kernel/cad_pid", mypid},
+		{"/proc/sys/kernel/cad_pid", tmppid},
+
 	};
 	for (size_t i = 0; i < sizeof(files) / sizeof(files[0]); i++) {
 		if (!write_file(files[i].name, files[i].data)) {
 			debug("write to %s failed: %s\n", files[i].name, strerror(errno));
 		}
 	}
+	kill(cad_pid, SIGKILL);
```

Comment:
Is it guaranteed that killing the process will not de facto unset `/proc/sys/kernel/cad_pid`? Could it be better to set PR_SET_PDEATHSIG in the child process and just let it run?

Reply:
cad_pid is only set from the sysctl:
https://elixir.bootlin.com/linux/latest/A/ident/cad_pid
and the variable holds `struct pid` reference, so we should be good:
https://elixir.bootlin.com/linux/latest/source/kernel/sysctl.c#L1344

Reply:
That's what we did before as well.